### ISSUE TITLE
How to test locally a trained policy by calling model directly ?

### ISSUE THREAD

**Issue by duburcqa** [ray-project/ray#13651]
I'm trying to test locally a custom policy trained using PPO, and having as view_requirements:
```
# Specify the inputs
self.view_requirements["prev_n_obs"] = ViewRequirement(
    data_col="obs",
    shift="-{}:0".format(num_frames - 1),  # Last included !
    space=obs_space)
self.view_requirements["prev_n_act"] = ViewRequirement(
    data_col="actions",
    shift="-{}:-1".format(num_frames),
    space=action_space)
self.view_requirements["prev_n_rewards"] = ViewRequirement(
    data_col="rewards",
    shift="-{}:-1".format(num_frames))
```

For now, I'm doing this:

```
# Get policy model
policy = trainer.get_policy()
model = policy.model
dist_class = policy.dist_class
obs_filter = trainer.workers.local_worker().filters['default_policy']

# Instantiate learning environment
env = trainer.env_creator(debug=True)

# Initialize frame stack
input_dict = {
    'obs': np.zeros([1, *env.observation_space.shape]),
    'prev_n_obs': np.zeros([1, num_frames, *env.observation_space.shape]),
    'prev_n_act': np.zeros([1, num_frames, *env.action_space.shape]),
    'prev_n_rewards': np.zeros([1, num_frames])
}

# Initialize the simulation
input_dict['obs'][0, :] = input_dict['prev_n_obs'][0, -1] = obs_filter(env.reset())

# Run the simulation
for _ in range(config["horizon"]):
    action_logits, _ = model(input_dict)
    action_dist = dist_class(action_logits, model)
    action_tf = action_dist.sample()  # sample()
    action = action_tf.eval(session=policy._sess)

    obs, reward, done, info = env.step(action)
    if done:
        break
    for k, v in input_dict.items():
        v[:] = np.roll(v, -1)
    input_dict['obs'][0, :] = input_dict['prev_n_obs'][0, -1] = obs_filter(obs)
    input_dict['prev_n_act'][0, -1] = action
    input_dict['prev_n_rewards'][0, -1] = reward
```

It is running without throwing an exception, but the behaviour is strange, and the reward in very low compared to the mean reward during training. Does someone have an idea of what I'm doing wrong ? I'm trying to avoid relying on black-box methods such as `compute_action` for several reasons, first it does not work with the `view_requirements` I'm specifying, and second want to use my model outside rllib so I need to understand the pre/post-processing steps that I need to perform on my data to evaluate the policy.

**Comment by duburcqa** [comment]
By the way, how I am supposed to initialize `input_dict` since for computing the action I need the observations, actions, and reward of the previous steps, while there is any... I cannot think to anything else than zero...

**Comment by duburcqa** [comment]
My bad, I found the issue, I was just doing the roll wrong. It should be instead: `np.roll(v, shift=-1, axis=1)`. Additionally, I did 
```
obs_filter = trainer.workers.local_worker().filters['default_policy']
obs_mean, obs_std = obs_filter.rs.mean, obs_filter.rs.std
obs_filter_fn = lambda obs: (obs - obs_mean) / (obs_std + 1e-8)
```
instead of using `obs_filter` directly since otherwise the testing data would be accumulated to update the filter, which is not what I want.
